cpp/graph_algorithms_module/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+set(CYCLES_MODULE cycles_module.cpp)
+
+set(SOURCES algorithms/biconnected_components.cpp
```

Comment:
Now, how would we split this? 
I think it's okay to split each algorithm into the separate folder, they already have their own modules.
Also, move the common stuff like graph and graph_interface to utils.